plugins/woocommerce/src/Internal/Traits/OrderAttributionMeta.php

```diff
@@ -381,48 +381,29 @@ private function get_field_description( string $field ): string {
 	/**
 	 * Get the order history for the customer (data matches Customers report).
 	 *
-	 * @param mixed $customer_identifier The customer ID or billing email.
+	 * @param int $customer_report_id The customer ID (not necessarily User ID).
 	 *
-	 * @return array Order count, total spend, and average spend per order.
+	 * @return array|null Order count, total spend, and average spend per order.
 	 */
-	private function get_customer_history( $customer_identifier ): array {
-		/*
-		 * Exclude the statuses that aren't valid for the Customers report.
-		 * 'checkout-draft' is the checkout block's draft order status. `any` is added by V2 Orders REST.
-		 * @see /Automattic/WooCommerce/Admin/API/Report/DataStore::get_excluded_report_order_statuses()
-		 */
-		$all_order_statuses = ReportsController::get_order_statuses();
-		$excluded_statuses  = array( 'pending', 'failed', 'cancelled', 'auto-draft', 'trash', 'checkout-draft', 'any' );
-
-		// Get the valid customer orders.
-		$args = array(
-			'limit'  => - 1,
-			'return' => 'objects',
-			'status' => array_diff( $all_order_statuses, $excluded_statuses ),
-			'type'   => 'shop_order',
-		);
-
-		// If the customer_identifier is a valid ID, use it. Otherwise, use the billing email.
-		if ( is_numeric( $customer_identifier ) && $customer_identifier > 0 ) {
-			$args['customer_id'] = $customer_identifier;
-		} else {
-			$args['billing_email'] = $customer_identifier;
-			$args['customer_id']   = 0;
-		}
-
-		$orders = wc_get_orders( $args );
-
-		// Populate the order_count and total_spent variables with the valid orders.
-		$order_count = count( $orders );
-		$total_spent = 0;
-		foreach ( $orders as $order ) {
-			$total_spent += $order->get_total() - $order->get_total_refunded();
-		}
+	private function get_customer_history( $customer_report_id ): ?array {
+		$matching_customers = $this->get_customer_reports_rest_response( array( 'customers' => $customer_report_id ) );
+		return $matching_customers[0] ?? null;
+	}
 
-		return array(
-			'order_count'   => $order_count,
-			'total_spent'   => $total_spent,
-			'average_spent' => $order_count ? $total_spent / $order_count : 0,
-		);
+	/**
+	 * Make a REST request to the customer reports (analytics) endpoint.
+	 *
+	 * @param array  $params The query parameters to use.
+	 * @param string $method The HTTP method to use, defaults to GET.
+	 *
+	 * @return array The server response data.
+	 *
+	 * @since 8.5.3
+	 */
+	private function get_customer_reports_rest_response( $params, $method = 'GET' ) {
+		$request = new \WP_REST_Request( 'GET', '/wc-analytics/reports/customers' );
```

Comment:
We should call the controller directly here, instead of going via the REST request. Current code will load the entirety of REST API  which makes it super slow and we are working to prevent it.

Reply:
Also looks like calling the reports controller for just customer order total is quite an overkill. If we'd like to show it on every admin page load, then we need to store it directly, perhaps by adding another column in `wc_customer_lookup`. Otherwise, maybe just a link to customer report page will suffice here?

Reply:
> Otherwise, maybe just a link to customer report page will suffice here?
That is worth considering but it would require input from desing and project management :)

Reply:
No worries, perhaps then we can cache the fields that we are interested in? (we can invalidate that cache on new orders for that customer)

Reply:
I've updated the process for retrieving the data to avoid using API calls. 3e12d28d9fdc0edd4faedef45eb76df9d1d52f86
The [customer history data is actually already cached](https://github.com/woocommerce/woocommerce/blob/5f00e2f87a1435bee0210b06c403ddea2ee6e2d2/plugins/woocommerce/src/Admin/API/Reports/Customers/DataStore.php#L437-L441) (for 7 days, I believe) when it's retrieved, so that isn't a priority.
However, adding new columns to `wc_customer_lookup` does seem like an interesting thing to consider for a future PR.